File: src/main/java/trap/model/IndividualTotal.java
package trap.model;

import trap.common.Classifications;

public record IndividualTotal(
        int locationId,
        String team,
        String athlete,
        String classification,
        String gender,
        int total,
        String type
) {

    public String teamForScores() {
        return type + " " + team + " " + teamClassificationForTotal();
    }

    public String teamClassification() {
        return switch (classification) {
            case "Senior/Varsity" -> Classifications.VARSITY;
            case "Senior/Jr. Varsity", Classifications.JUNIOR_VARSITY -> Classifications.JUNIOR_VARSITY;
            case "Intermediate/Advanced" -> Classifications.INTERMEDIATE_ADVANCED;
            case "Intermediate/Entry Level" -> Classifications.INTERMEDIATE_ENTRY;
            default -> classification;
        };
    }

    public String teamClassificationForTotal() {
        return switch (teamClassification()) {
            case "Senior/Varsity", "Senior/Jr. Varsity", Classifications.VARSITY, Classifications.JUNIOR_VARSITY -> Classifications.VARSITY;
            case Classifications.INTERMEDIATE_ADVANCED -> Classifications.INTERMEDIATE_ENTRY;
            default -> teamClassification();
        };
    }
}

File: src/main/java/trap/model/IndividualTotal.kt
package trap.model

class IndividualTotal(val locationId: Int, val team: String, val athlete: String, val classification: String, val gender: String, val total: Int, val type: String) {

    val teamForScores: String
        get() = "$type $team $teamClassificationForTotal"
    val teamClassification: String
        get() = classification.replace("Senior/Varsity", "Varsity").replace("Senior/Jr. Varsity", "Junior Varsity").replace("Intermediate/Advanced", "Intermediate Advanced").replace("Intermediate/Entry Level", "Intermediate Entry")
    val teamClassificationForTotal: String
        get() = teamClassification.replace("Senior/Jr. Varsity", "Varsity").replace("Senior/Varsity", "Varsity").replace("Junior Varsity", "Varsity").replace("Intermediate Advanced", "Intermediate Entry")
}
